### ISSUE TITLE
liblo not found on OS X 10.9

### ISSUE THREAD

**Issue by alesaccoia** [monome/libmonome#39]
Hi, I have been trying to build libmonome with the following liblo:
- homebrew version (just dynamic)
- compiled from source (fat lib, static + dynamic)

In no way waf will find it. I printed "conf.env.LIBDIR" and it's actually pointing to the right place, /usr/local/lib. adding -vvv to configure doesn't really make a difference because it doesn't show the error code of clang. what could it be? I am on Mavericks.. cheers

**Comment by wrl** [comment]
could you paste me the results of `build/config.log`?

**Comment by alesaccoia** [comment]
here it is

# project libmonome (1.4.0) configured on Tue Aug  4 02:52:34 2015 by
# waf 1.8.5 (abi 98, python 20709f0 on darwin)
# using ./waf configure
#
----------------------------------------
Setting top to
/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome
----------------------------------------
Setting out to
/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/build
----------------------------------------
Checking for 'clang' (C compiler)
find program=['clang'] paths=['/Volumes/Data/ANDROID/android-ndk-r9d', '/Volumes/Data/ANDROID/android-sdk-macosx/tools', '/Volumes/Data/ANDROID/android-sdk-macosx/platform-tools', '/Volumes/Data/ANDROID/android-sdk-macosx/build-tools/22.0.1/', '/Applications/gradle-2.4/bin/', '/usr/local/bin', '/usr/local/sbin', '/opt/local/bin', '/opt/local/sbin', '/Volumes/Data/Development/Javascript/depot_tools', '/usr/local/mysql/bin', '/Users/alex/cdpr7/_cdp/_cdprogs', '', '/opt/local/bin', '/opt/local/sbin', '/usr/bin', '/bin', '/usr/sbin', '/sbin', '/usr/local/bin', '/Applications/Xcode.app/Contents/Developer/usr/bin', '/opt/X11/bin', '/usr/local/MacGPG2/bin', '/usr/texbin'] var='CC' -> ['/usr/bin/clang']
find program=['ar'] paths=['/Volumes/Data/ANDROID/android-ndk-r9d', '/Volumes/Data/ANDROID/android-sdk-macosx/tools', '/Volumes/Data/ANDROID/android-sdk-macosx/platform-tools', '/Volumes/Data/ANDROID/android-sdk-macosx/build-tools/22.0.1/', '/Applications/gradle-2.4/bin/', '/usr/local/bin', '/usr/local/sbin', '/opt/local/bin', '/opt/local/sbin', '/Volumes/Data/Development/Javascript/depot_tools', '/usr/local/mysql/bin', '/Users/alex/cdpr7/_cdp/_cdprogs', '', '/opt/local/bin', '/opt/local/sbin', '/usr/bin', '/bin', '/usr/sbin', '/sbin', '/usr/local/bin', '/Applications/Xcode.app/Contents/Developer/usr/bin', '/opt/X11/bin', '/usr/local/MacGPG2/bin', '/usr/texbin'] var='AR' -> ['/usr/bin/ar']
/usr/bin/clang
----------------------------------------
Checking for endianness
==>

short int ascii_mm[] = { 0x4249, 0x4765, 0x6E44, 0x6961, 0x6E53, 0x7953, 0 };
short int ascii_ii[] = { 0x694C, 0x5454, 0x656C, 0x6E45, 0x6944, 0x6E61, 0 };
int use_ascii (int i) {
	return ascii_mm[i] + ascii_ii[i];
}
short int ebcdic_ii[] = { 0x89D3, 0xE3E3, 0x8593, 0x95C5, 0x89C4, 0x9581, 0 };
short int ebcdic_mm[] = { 0xC2C9, 0xC785, 0x95C4, 0x8981, 0x95E2, 0xA8E2, 0 };
int use_ebcdic (int i) {
	return ebcdic_mm[i] + ebcdic_ii[i];
}
extern int foo;

<==
[1/2] Compiling [32mbuild/.conf_check_484ef2a953e9f19f01dd990244e32cfa/test.c[0m

['/usr/bin/clang', '../test.c', '-c', '-o', '/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/build/.conf_check_484ef2a953e9f19f01dd990244e32cfa/testbuild/test.c.1.o']
[2/2] Processing [35mbuild/.conf_check_484ef2a953e9f19f01dd990244e32cfa/testbuild/test.c.1.o[0m

little
----------------------------------------
Checking for working poll()
==>

		#include <stdlib.h>
		#include <poll.h>

		int main(int argc, char **argv) {
		    struct pollfd fds[1];

		    fds[0].fd = open("/dev/null", 1);
		    fds[0].events = POLLIN;

		    if( poll(fds, 1, 0) < 0 || fds[0].revents & POLLNVAL )
		        exit(1);
		    exit(0);
		}
<==
[1/3] Compiling [32mbuild/.conf_check_a8fbdb3731a6d9952122e6427a9dd853/test.c[0m

['/usr/bin/clang', '../test.c', '-c', '-o', '/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/build/.conf_check_a8fbdb3731a6d9952122e6427a9dd853/testbuild/test.c.1.o']
err: ../test.c:8:19: warning: implicit declaration of function 'open' is invalid in C99 [-Wimplicit-function-declaration]
                    fds[0].fd = open("/dev/null", 1);
                                ^
1 warning generated.

[2/3] Linking [33mbuild/.conf_check_a8fbdb3731a6d9952122e6427a9dd853/testbuild/testprog[0m

['/usr/bin/clang', 'test.c.1.o', '-o', '/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/build/.conf_check_a8fbdb3731a6d9952122e6427a9dd853/testbuild/testprog']
[3/3] Processing [35mbuild/.conf_check_a8fbdb3731a6d9952122e6427a9dd853/testbuild/testprog[0m

['/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/build/.conf_check_a8fbdb3731a6d9952122e6427a9dd853/testbuild/testprog']
no (will use select())
from /Volumes/Data/code/nvstudio/monome/thirdparty/libmonome: The configuration failed False
----------------------------------------
Checking for library dl
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/2] Compiling [32mbuild/.conf_check_1d65952fba078f4d8a17d901caa4cd39/test.c[0m

['/usr/bin/clang', '../test.c', '-c', '-o', '/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/build/.conf_check_1d65952fba078f4d8a17d901caa4cd39/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_1d65952fba078f4d8a17d901caa4cd39/testbuild/testprog[0m

['/usr/bin/clang', 'test.c.1.o', '-o', '/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/build/.conf_check_1d65952fba078f4d8a17d901caa4cd39/testbuild/testprog', '-ldl']
yes
----------------------------------------
Checking for liblo
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/3] Compiling [32mbuild/.conf_check_7df6e36047bd8ce78bbfe7ba013e3a95/test.c[0m

['/usr/bin/clang', '../test.c', '-c', '-o', '/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/build/.conf_check_7df6e36047bd8ce78bbfe7ba013e3a95/testbuild/test.c.1.o']
[2/3] Linking [33mbuild/.conf_check_7df6e36047bd8ce78bbfe7ba013e3a95/testbuild/testprog[0m

['/usr/bin/clang', 'test.c.1.o', '-o', '/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/build/.conf_check_7df6e36047bd8ce78bbfe7ba013e3a95/testbuild/testprog', '-llo']
err: ld: library not found for -llo
clang: error: linker command failed with exit code 1 (use -v to see invocation)

from /Volumes/Data/code/nvstudio/monome/thirdparty/libmonome: Test does not build: Traceback (most recent call last):
  File "/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/.waf-1.8.5-4caa3bbe43c4e50e4effc29787929b8f/waflib/Configure.py", line 329, in run_build
    bld.compile()
  File "/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/.waf-1.8.5-4caa3bbe43c4e50e4effc29787929b8f/waflib/Build.py", line 182, in compile
    raise Errors.BuildError(self.producer.error)
BuildError: Build failed
 -> task in 'testprog' failed (exit status 1): 
	{task 4356878992: cprogram test.c.1.o -> testprog}
['/usr/bin/clang', 'test.c.1.o', '-o', '/Volumes/Data/code/nvstudio/monome/thirdparty/libmonome/build/.conf_check_7df6e36047bd8ce78bbfe7ba013e3a95/testbuild/testprog', '-llo']

not found
from /Volumes/Data/code/nvstudio/monome/thirdparty/libmonome: The configuration failed

**Comment by wrl** [comment]
man. so, here's a weird question: have you rebooted between the time that you installed liblo and tried to build libmonome? in my experience, it takes a reboot to get OSX to rebuild the dyld cache and start using it.

worth a shot.

**Comment by alesaccoia** [comment]
rebooted now, it still doesn't work.. I have no clue

**Comment by alesaccoia** [comment]
ls /usr/local/lib/liblo.
liblo.0.6.0.dylib  liblo.0.dylib      liblo.7.dylib      liblo.a            liblo.dylib        liblo.la

lipo -detailed_info /usr/local/lib/liblo.a
Fat header in: /usr/local/lib/liblo.a
....

same for the dylib

**Comment by alesaccoia** [comment]
Wow I have spent so many hours on this.
libmonome doesn't compile for me... what I did, I have explicitly moved these to line 138, before check_liblo. Notice that I have explicitly added the -L and -I flags. now liblo is found

     conf.env.append_unique("CFLAGS", [
	  "-I/usr/local/include",
		"-std=c99", "-Wall", "-Werror",
		"-Wno-initializer-overrides"])
	
	conf.env.append_unique("LINKFLAGS", [
		"-L/usr/local/lib"])

But when compiling, a function declaration in monome.h is missing when compiling src/monomeserial.c. Log here

http://pastebin.com/i0zjPTVV

So I installed libmonome from homebrew :-/. Weird that when using
brew install --build-from-source libmonome 
everything works (it's calling **./waf configure --prefix=/usr/local/Cellar/libmonome/1.2** )
For serialosc, I have been able to compile it, after doing the same thing: adding -I and -L and moving it higher in the wscript.
Still no idea... and it's the first time I use waf
